homeassistant/components/garmin_connect/__init__.py

```diff
@@ -66,38 +77,24 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
     return unload_ok
 
 
-class GarminConnectData:
-    """Define an object to hold sensor data."""
-
-    def __init__(self, hass, client):
-        """Initialize."""
-        self.hass = hass
-        self.client = client
-        self.data = None
-
-    @Throttle(DEFAULT_UPDATE_INTERVAL)
-    async def async_update(self):
-        """Update data via API wrapper."""
-        today = date.today()
-
-        try:
-            summary = await self.client.get_user_summary(today.isoformat())
-            body = await self.client.get_body_composition(today.isoformat())
-
-            self.data = {
-                **summary,
-                **body["totalAverage"],
-            }
-            self.data["nextAlarm"] = await self.client.get_device_alarms()
-        except (
-            GarminConnectAuthenticationError,
-            GarminConnectTooManyRequestsError,
-            GarminConnectConnectionError,
-        ) as err:
-            _LOGGER.error(
-                "Error occurred during Garmin Connect update requests: %s", err
-            )
-        except Exception:  # pylint: disable=broad-except
-            _LOGGER.exception(
-                "Unknown error occurred during Garmin Connect update requests"
-            )
+async def async_update_garmin_data(api):
+    """Fetch data from API endpoint."""
+    try:
+        summary = await api.get_user_summary(date.today().isoformat())
+        body = await api.get_body_composition(date.today().isoformat())
+        alarms = await api.get_device_alarms()
+
+        return {
```

Comment:
Please only wrap the line(s) that can raise in the try... except block. Move the rest down.